tests/integration/datastore/test_datastore.py

```diff
@@ -14,64 +14,392 @@
 You should have received a copy of the GNU Affero General Public License
 along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
 """
+import lmdb
+import maya
+import msgpack
 import pytest
+import tempfile
 from datetime import datetime
 
 from nucypher.datastore import datastore, keypairs
+from nucypher.datastore.base import DatastoreRecord, RecordField
+from nucypher.datastore.models import PolicyArrangement, Workorder
 
+class TestRecord(DatastoreRecord):
+    _test = RecordField(bytes)
+    _test_date = RecordField(datetime,
+            encode=lambda val: datetime.isoformat(val).encode(),
+            decode=lambda val: datetime.fromisoformat(val.decode()))
 
-@pytest.mark.usefixtures('testerchain')
-def test_key_sqlite_datastore(test_datastore, federated_bob):
 
-    # Test add pubkey
-    test_datastore.add_key(federated_bob.stamp, is_signing=True)
+def test_datastore_describe():
+    temp_path = tempfile.mkdtemp()
+    storage = datastore.Datastore(temp_path)
+    assert storage.LMDB_MAP_SIZE == 1_000_000_000_000
+    assert storage.db_path == temp_path
+    assert storage._Datastore__db_env.path() == temp_path
 
-    # Test get pubkey
-    query_key = test_datastore.get_key(federated_bob.stamp.fingerprint())
-    assert bytes(federated_bob.stamp) == bytes(query_key)
+    #
+    # Tests for `Datastore.describe`
+    #
 
-    # Test del pubkey
-    test_datastore.del_key(federated_bob.stamp.fingerprint())
-    with pytest.raises(datastore.NotFound):
-        del_key = test_datastore.get_key(federated_bob.stamp.fingerprint())
+    # Getting writeable access to a record can be done by not setting `writeable` to `True`.
+    # `writeable` is, by default, `False`.
```

Comment:
Doesn't look like "not" should be there.

Reply:
haha good looking out! Thanks!